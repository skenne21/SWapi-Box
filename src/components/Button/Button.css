#selected {
  background-color: #BFA514;
}

.favs-button {
  width: 80%;
  height: 50px;
  margin-top: 15px;
  font-size: 1.4em;
}

.favs-card {
  padding: 2px;
  background-color: #6B2427; 
  font-size: 3em;
  width: 20%;
  height: 60px;
  border-radius: 50%;
  border: none;
}

.Button {
  margin: 25px;
  width: 200px;
  height: 60px;
  font-size: 2em;
  background-color: #FFD700;

}