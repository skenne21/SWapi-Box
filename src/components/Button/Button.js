import React from 'react';
import PropTypes from 'prop-types';
import './Button.css';

const Button = ({className, name, controlFunc, isActive, card = {}}) => {
  const selected = () => {
    if (isActive === name) {
      return true;
    }
    if (card.id !== undefined) {
      return true;
    }
  };

  return (
    <div>
      <button
        className={className}
        id={selected() ? 'selected' : ''}
        onClick={(() => { controlFunc(name, card)}) }>{name}</button> 
    </div>
  );
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  controlFunc: PropTypes.func, 
  isActive: PropTypes.string, 
  card: PropTypes.object
};

export default Button;

