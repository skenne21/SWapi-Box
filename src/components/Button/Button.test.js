import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button.js';

describe('Button', () => {
  let wrapper, mockControlFunc, name, card;

  beforeEach(() => {
    mockControlFunc = jest.fn();
    name='People';
    card = {};
    wrapper = shallow(
      <Button 
        name={name} 
        controlFunc={mockControlFunc}
        isActive={''}
        card={card}
      />);
  });

  it('Should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call controlFunc when clicked', () => {
    wrapper.find('button').simulate('click');
    expect(mockControlFunc).toHaveBeenCalledWith(name, card);
  });

  it('Should had an ID of selected if active = name passed', () => {
    wrapper = shallow(<Button 
      name={name} 
      controlFunc={mockControlFunc}
      isActive={'People'}
      card={card}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should had an ID of selected if card.id is undefined', () => {
    const card = {
      id: undefined
    };
    wrapper = shallow(<Button 
      name={name} 
      controlFunc={mockControlFunc}
      isActive={'People'}
      card={card}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});