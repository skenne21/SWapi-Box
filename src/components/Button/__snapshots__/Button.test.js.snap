// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match the snapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    card={Object {}}
    controlFunc={[Function]}
    isActive=""
    name="People"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className={undefined}
        id=""
        onClick={[Function]}
>
        People
</button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "People",
        "className": undefined,
        "id": "",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "People",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className={undefined}
          id=""
          onClick={[Function]}
>
          People
</button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": undefined,
          "id": "",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
